bindings/python/tests/test_parse.py

```diff
@@ -135,3 +135,124 @@ def test_parm_decl(tmp_path):
     assert parm_decl["kind"] == "PARM_DECL"
     assert parm_decl["element_type"] == "Int"
     assert parm_decl["name"] == "aFunctionParameter"
+
+
+def test_struct_decl(tmp_path):
+    file_contents = "struct AStruct {};"
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+
+    assert struct_decl["kind"] == "STRUCT_DECL"
+    assert struct_decl["name"] == "AStruct"
+
+
+def test_public_inheritance(tmp_path):
+    file_contents = """
+    struct BaseStruct {};
+    struct DerivedStruct: public BaseStruct {};
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    child_struct_decl = parsed_info["members"][1]
+    cxx_base_specifier = child_struct_decl["members"][0]
+
+    assert cxx_base_specifier["kind"] == "CXX_BASE_SPECIFIER"
+    assert cxx_base_specifier["access_specifier"] == "PUBLIC"
+    assert cxx_base_specifier["name"] == "struct BaseStruct"
+
+
+def test_member_function(tmp_path):
+    file_contents = """
+    struct AStruct {
+        void aMethod() {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    cxx_method = struct_decl["members"][0]
+
+    assert cxx_method["kind"] == "CXX_METHOD"
+    assert cxx_method["result_type"] == "void"
+    assert cxx_method["name"] == "aMethod"
+
+
+def test_type_ref(tmp_path):
+    file_contents = """
+    struct SomeUsefulType {};
+
+    class AClass {
+        void aMethod(SomeUsefulType aParameter) {};
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    class_decl = parsed_info["members"][1]
+    cxx_method = class_decl["members"][0]
+    parm_decl = cxx_method["members"][0]
+    type_ref = parm_decl["members"][0]
+
+    assert type_ref["kind"] == "TYPE_REF"
+    assert type_ref["name"] == "struct SomeUsefulType"
+
+
+def test_simple_constructor(tmp_path):
+    file_contents = """
+    struct AStruct {
+        AStruct() {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    constructor = struct_decl["members"][0]
+
+    assert constructor["kind"] == "CONSTRUCTOR"
+    assert constructor["access_specifier"] == "PUBLIC"
+    assert constructor["name"] == "AStruct"
```

Comment:
LGTM. Will the test for default ctor and defaulted/deleted ctor be separate?

Reply:
New PR: #30